src/session/session_api.c

```diff
@@ -804,27 +804,33 @@ __wt_session_drop(WT_SESSION_IMPL *session, const char *uri, const char *cfg[])
 	WT_RET(__wt_config_gets_def(session, cfg, "checkpoint_wait", 1, &cval));
 	checkpoint_wait = cval.val != 0;
 	WT_RET(__wt_config_gets_def(session, cfg, "lock_wait", 1, &cval));
-	lock_wait = cval.val != 0 || F_ISSET(session, WT_SESSION_LOCK_NO_WAIT);
-
-	if (!lock_wait)
-		F_SET(session, WT_SESSION_LOCK_NO_WAIT);
```

Comment:
@michaelcahill, notice I removed the `F_ISSET(session, WT_SESSION_LOCK_NO_WAIT);` part of the `lock_wait` test. I couldn't figure out what that did, it just didn't make any sense to me.